File: Linguagem C/while/ex_012.c
//12) Faça um programa que leia um número N e mostre a soma dos N primeiros termos da seguinte
//expressão: S = 1 – 2 + 3 – 4 + 5 – 6 + ... + N.
#include <stdio.h>
int main (){
    int n, soma=0, i=1, sinal=1;
    scanf("%d", &n);

    while(i <= n){
        soma +=i*sinal;
        i++;
        sinal*=-1;
    }
    printf("%d\n", soma);
}

File: Python/005_Lista_While/Ex_012.py
"""
12)  Faça  um  programa  que  leia  um  número  N  e  mostre  a  soma  dos  N  primeiros  termos  da  seguinte 
expressão: S = 1 – 2 + 3 – 4 + 5 – 6 + ... + N.
"""
n1 = int(input("Digite um numero para n1: "))

sinal = 1
soma = 0
for num in range(1, n1+1, 1):
    soma += sinal*num
    sinal*= -1
print(soma)